### PR TITLE
Add Comparable conformance to HTTPVersion

### PR DESCRIPTION
**Pull request by seabaylea**
Add Comparable support for HTTPVersion plus associated tests.

### CODE REVIEW

File: Sources/HTTP/HTTPVersion.swift [hunk 1]
```diff
@@ -29,12 +29,44 @@ extension HTTPVersion : Hashable {
...
+            return lhs.minor < rhs.minor
+        }
+    }
+    
+    public static func <= (lhs: HTTPVersion, rhs: HTTPVersion) -> Bool {
+        return lhs < rhs || lhs == rhs
+    }
+
+}
+
```
Comment (aleksey-mashanov) [inline Sources/HTTP/HTTPVersion.swift]: Implementing `<` is enough for conformance to `Comparable`. Other operators are provided by stdlib in extension to `Comparable`. I don't see any performance advantages in implementing all operators by hand.

File: Sources/HTTP/HTTPVersion.swift [hunk 2]
```diff
@@ -29,12 +29,44 @@ extension HTTPVersion : Hashable {
     public static func == (lhs: HTTPVersion, rhs: HTTPVersion) -> Bool {
         return lhs.hashValue == rhs.hashValue
     }
-
```
Comment (aleksey-mashanov) [inline Sources/HTTP/HTTPVersion.swift]: Hash values by design are guaranteed to be equal for two equal values but not vice versa.
So it is better to replace implementation of `==` with `lhs.major == rhs.major && lhs.minor == rhs.minor`.

For example, in current implementation hash values for versions 1.1 and 2.2 are equal. I suggest to implement `hashValue` in a more collision-resistant way, something like: `(major << 8) | minor`

Comment (michael-yuji) [reply Sources/HTTP/HTTPVersion.swift]: exactly.

File: Sources/HTTP/HTTPVersion.swift (hunk 1 above)
Comment (seabaylea) [reply Sources/HTTP/HTTPVersion.swift]: Agreed - looking at the implementation in the stdlib:
https://github.com/apple/swift/blob/767e8f31ac8ab8186102dc95fe5916bb2d29bbb5/stdlib/public/core/Comparable.swift
there's no advantage to implementing these by hand. I'll update to remove.

File: Sources/HTTP/HTTPVersion.swift (hunk 2 above)
Comment (seabaylea) [reply Sources/HTTP/HTTPVersion.swift]: I'll update to remove the potential hash collision and replace the implementation on `==`

File: Sources/HTTP/HTTPVersion.swift [hunk 3]
```diff
@@ -21,20 +21,34 @@ public struct HTTPVersion {
...
+        return (major << 8) | minor
     }
 
     public static func == (lhs: HTTPVersion, rhs: HTTPVersion) -> Bool {
-        return lhs.hashValue == rhs.hashValue
+        return lhs.major == rhs.major && lhs.minor == rhs.minor
     }
-
+    
+    public static func != (lhs: HTTPVersion, rhs: HTTPVersion) -> Bool {
```
Comment (aleksey-mashanov) [inline Sources/HTTP/HTTPVersion.swift]: `!=` should be removed too.

Comment (seabaylea) [reply Sources/HTTP/HTTPVersion.swift]: Done.

**Comment by seabaylea** [comment]
If there's no further review comments, I'll look to merge this tomorrow.

**Comment by seabaylea** [comment]
Merging as there's no more review comments.

**Pull request merged**